### DISCUSSION TITLE
Repeatable: Flexible Content

### DISCUSSION THREAD

**Discussion by ericjanofski** [Laravel-Backpack/community-forum#987] [Q&A (Help)]
Hey there, I was wondering if there's some way to setup different templates for the repeatable field or if a similar field is in place.

I'm constantly in need of this features.

For instance, on a current project I'd love for the user to be able to click a button and be able to add any amount of content that falls under different field templates such as a media file, a quiz, a paragraph with a heading and a uploaded file. 

What I have to do instead is setup a crud for each of these types and then have a bunch of hidden fields and a type field that toggles them on and off. Then I have to handle the logic in the api that grabs the correct field content. It's very tedious and prone to error.

The closes example I can think of is the ACF Flexible Content Field, which is top knotch.

Let me know if something like that exists or if it's in the works. Thanks!!!

**Comment by karandatwani92** [comment]
Hey @ericjanofski 


IDK if the way I'd do it is the same as you said. But here is the way I would do it:

1. Create the first subfield as `type_of_field` in the [repeatable field](https://backpackforlaravel.com/docs/6.x/crud-fields#repeatable-pro) and add all fields in subfields mentioned in `type_of_field` dropdown.
2. [Use CrudField JavaScript Library](https://backpackforlaravel.com/docs/6.x/crud-fields-javascript-api) to hide other fields of the row and show only the selected one.
3. Information will be stored in JSON. So, I'd not run any alteration on saving. I'll use stored JSON to render.

**Comment by karandatwani92** [comment]
If you think ACF Flexible Content Field should be a ready field. You can add a feature request and vote for it on
[What features should we build next?](https://github.com/Laravel-Backpack/community-forum/discussions/751)

**Comment by ericjanofski** [comment]
Your solution is exactly what I'm doing. It works but it's a bit clunky and tedious and since there are so many steps it's prone to error on my part. I will certainly suggest the flexible content setup. Cristian had a soluion in place similar to what I'm requesting in v4 I believe. It worked great for my needs, perhaps it can be revived! :) Thank you for your help!
